Automattic/jetpack
# Dashboard: Fix all React warnings shown on the console

Fixes annoying warnings and more.

#### Changes proposed in this Pull Request:

* Does a bunch of changes like fixing types declared `Proptypes` for some components or adding `key` props to looped renderings.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* Check this branch out.
* Enable SCRIPT_DEBUG
* build Jetpack  with `jetpack build plugins/jetpack`.
* Visit the admin page . 
* Open the console
* Confirm you only see these red warnings:
  *  The one tracked in #22516
  *  One about usage of `javascript:void(0)` in `href` props for an `a`. Will handle this on another PR


##### How to assert each of the commits is fine

_I'm still working on this list_

* **Commit Add key prop to security bandle dash info feature rendering:**
  Remember that a unique key prop is needed when looping in react component.
* **Commit Add PLAN_JETPACK_SECURITY_REALTIME_MONTHLY to accepted proptypes for…** 
  Check that we were rendering a videopress plan icon but the `jetpack_videopress` plan was not listed among possible values for that property `plan` from the PlanIcon component.
* **Commit Update string prop to boolean in Banner component**
  Check that the only use of `isCurrentUserLinked ` in the component is on boolean checks
* **Commit Update string prop to boolean in DashSearch AtAGlance component**
  Check that the only use of `hasConnectedOwner ` in the component is on boolean checks
* **Commit Update string prop to boolean in SupportCard component**
  Check that the only use of `isCurrentUserLinked ` in the component is on boolean checks
* **Commit Update title props in Banner and JetpackBanner to be node as we are c…** 
   `node` is a type that accepts both strings and elements. 
* **Commit Fix VideoPress dash component prop type name. Rename from isOfflineMo…** 
  Confirm that on that component there's no sue of a prop like isOfflineMode. That's the name of the state selector. The proper prop name is `isOffline`.
* **Commit Remove href with javascript URL from button in Apps Card**
  This href prop with this value is just unnecessary.
